x-pack/plugins/security_solution/public/common/components/visualization_actions/use_visualization_response.tsx

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useMemo } from 'react';
+
+import { parseVisualizationData } from './utils';
+import { useDeepEqualSelector } from '../../hooks/use_selector';
+import { inputsSelectors } from '../../store';
+import type { VisualizationResponse } from './types';
+
+export const useVisualizationResponse = ({ visualizationId }: { visualizationId: string }) => {
```

Comment:
this is a nice hook. I think you can replace some code you have in [`visualization_embeddable.tsx`](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/public/common/components/visualization_actions/visualization_embeddable.tsx#L42-L46)